cmake/pcl_find_vtk.cmake

```
@@ -119,6 +119,15 @@ else()
   unset(HAVE_QVTK)
 endif()
 
+# Overwrite VTK_LIBRARIES with only the set we actually want for VTK >= 9.0.
+# Otherwise, it will contain ALL available components.
+if(NOT (VTK_VERSION VERSION_LESS 9.0))
+  set(VTK_LIBRARIES)
+  foreach(vtkComponent ${PCL_VTK_COMPONENTS})
+    list(APPEND VTK_LIBRARIES VTK::${vtkComponent})
+  endforeach()
+endif()
```

Would it be possible to confirm that our `VTK_LIBRARIES` are a subset of the original?```suggestion
  set(ORIGINAL_VTK_LIBRARIES $VTK_LIBRARIES})
  set(REQUIRED_VTK_LIBRARIES)
  foreach(vtkComponent ${PCL_VTK_COMPONENTS})
    list(APPEND REQUIRED_VTK_LIBRARIES VTK::${vtkComponent})
  endforeach()
  # TODO: ensure that REQUIRED_VTK_LIBRARIES is subset of ORIGINAL_VTK_LIBRARIES
  set(VTK_LIBRARIES ${REQUIRED_VTK_LIBRARIES})
```